snuba/datasets/storages/factory.py

```diff
@@ -63,19 +64,14 @@
 
 USE_CONFIG_BUILT_STORAGES = "use_config_built_storages"
 
-CONFIG_FILES = {
-    StorageKey.GENERIC_METRICS_DISTRIBUTIONS: f"{settings.STORAGE_CONFIG_FILES_PATH}/distributions.yaml",
-    StorageKey.GENERIC_METRICS_DISTRIBUTIONS_RAW: f"{settings.STORAGE_CONFIG_FILES_PATH}/distributions_bucket.yaml",
-    StorageKey.GENERIC_METRICS_SETS_RAW: f"{settings.STORAGE_CONFIG_FILES_PATH}/sets_bucket.yaml",
-    StorageKey.GENERIC_METRICS_SETS: f"{settings.STORAGE_CONFIG_FILES_PATH}/sets.yaml",
-}
-
 CONFIG_BUILT_STORAGES = {
-    storage_key: build_storage(CONFIG_FILES[storage_key])
-    for storage_key in CONFIG_FILES
+    storage.get_storage_key(): storage
+    for storage in [
+        build_storage(config_file)
```

Comment:
What happens here if a non-storage config file ends up in the folder?

Reply:
The glob function specifically looks for the pattern `/storages/*.yaml`. So if there is a yaml file under a folder named `storages` in any subfolder of `datasets/configuration/`, it _should_ be a storage. If it isn't, the validation or building the actual storage will raise exceptions well before anything is merged to master.